### ISSUE TITLE
Consumable check - Vantus Rune Sepulcher of the First Ones

### ISSUE THREAD

**Issue by fraerie** [BigWigsMods/oRA3#109]
### What steps will reproduce the problem?

1.  Check Buffs
2.  view Vantus Rune column
3.  status of Sepulcher of the First Ones vasntus rune not displayed

### What version of the addon are you using? (Stating 'latest' is not useful)
9.2.1 for game version 9.2.5

### Any additional information? (example: WoW language if not English)
<img width="416" alt="ORA3 9 2 1 consumables" src="https://user-images.githubusercontent.com/76411103/171355031-8c8775f0-ca7f-4b8e-baf5-eafc08ea022c.png">

**Comment by TreizeEU** [comment]
Hello.

Are there a fix in the works for this? I only noticed it wasn't working today, so I thought if I had missed an update.

While trying to fix this myself, I encountered a weird issue with wowhead having duplicates of spells for each of the bosses in Sepulcher with different spell ID's. I tried to add each boss with one of the ID's and encounter ID's from the below webpage, which doesn't make it track for the Jailer's vantus rune spell ID. Any ideas? I hope any of this helps you figure out a fix for this?

https://wowpedia.fandom.com/wiki/DungeonEncounterID


Code I attempted to add to line 64 in file \oRA3\modules\Consumables.lua:


		-- Sepulcher of the First Ones
		[354403] = 2512, -- Guardian of the First Ones
		[367120] = 2542, -- Skolex, the Insatiable Ravener
		[367124] = 2553, -- Artificer Xy'mox
		[367125] = 2540, -- Dausegne, the Fallen Oracle
		[367127] = 2544, -- Prototype Pantheon
		[367129] = 2539, -- Lihuvim, Principal Architect
		[367131] = 2529, -- Halondrus the Reclaimer
		[367133] = 2546, -- Anduin Wrynn
		[367135] = 2543, -- Lords of Dread
		[367139] = 2549, -- Rygelon
		[367143] = 2537, -- The Jailer, Zovaal
	}

**Comment by fraerie** [comment]
As far as I can tell, the two IDs are one that sets the flag that a vantus rune has been used - so you can't use another one that week. The second one is applying the aura which gives the bonus versatility against that boss.

So for Guardian of the First Ones, the vantus rune used flag is spell ID 354403, and the aura is spell ID 354390.

What I can't work out is what the second number is - the '2458' for Guardian in your list.

**Comment by TreizeEU** [comment]
Hello fraerie.

I've worked out the vantus rune code for the jailer. I took the second number from the JournalEncounterID page I linked, then swapped the first number with the two versions, and landed on it working with 367143 (since I have an active vantus rune for jailer on my main). I'm not sure if it's the same ID (aura as opposed to flag) for all of them?

**Comment by TreizeEU** [comment]
Also, I helped fixing issue #108 while I was at it, same file.

**Comment by fraerie** [comment]
I've gone through and cross checked the encounter IDs from the link you posted (the encounter ID you had for Artificer was incorrect), and checked the current Vantus rune spell ids from the ORA3 code against the flag vs aura IDs in the WowHead database. I think it's the Aura Applied spell ID.

I've edited my consumables file - I'll let you know if it works this week.

**Comment by TreizeEU** [comment]
Oh yeah, I just noticed I had duped the number for first boss with Artificer. Artificer's second number should be 2470, indeed.

**Comment by nebularg** [comment]
If you want to make some pull requests I'll merge them in :p I'll probably take a look at oRA and update stuff sometime later this week

**Comment by fraerie** [comment]
> If you want to make some pull requests I'll merge them in :p I'll probably take a look at oRA and update stuff sometime later this week

Um...I'm not a dev so I don't really know what a pull request involves... I'm a BA and can cross check tables of data, but doing pull requests is a bit outside my wheelhouse. :)

**Comment by TreizeEU** [comment]
I put up a pull request for the Eternal Augment Rune. I know that one works. Wasn't sure if I could add both changes (Vantus rune and Augment rune) in the same pull request, and wasn't sure how much of the Vantus rune code actually works.

**Comment by fraerie** [comment]
I may be able to test Jailer or Rygalon this week depending on how progression goes.

**Comment by TreizeEU** [comment]
This is my current Consumables.lua file:

[Consumables.zip](https://github.com/BigWigsMods/oRA3/files/9052350/Consumables.zip)

**Comment by TreizeEU** [comment]
I didn't create a pull request, I made a fork instead because I don't know what I'm doing. XD

**Comment by fraerie** [comment]
> If you want to make some pull requests I'll merge them in :p I'll probably take a look at oRA and update stuff sometime later this week

Both Treize and I looked at creating a pull request and chickened out, but have verified the consumables file linked above works and is stable.

**Comment by fraerie** [comment]
Re the message I got saying this had already been fixed - I've checked the code for the version currently on the DL link (v9.2.1), it only includes the following runes

```
local getVantus, getVantusBoss
do
	local runes = {
		-- Castle Nathria
		[311445] = 2393, -- Shriekwing
		[311446] = 2428, -- Hungering Destroyer
		[311448] = 2422, -- Sun King's Salvation
		[311447] = 2418, -- Artificer Xy'mox
		[311449] = 2420, -- Lady Inerva Darkvein
		[311450] = 2426, -- The Council of Blood
		[311451] = 2394, -- Sludgefist
		[311452] = 2425, -- Stone Legion Generals
		[334131] = 2424, -- Sire Denathrius
		[334132] = 2429, -- Huntsman Altimor
		-- Sanctum of Domination
		[354384] = 2435, -- The Tarragrue
		[354385] = 2442, -- The Eye of the Jailer
		[354386] = 2439, -- The Nine
		[354387] = 2444, -- Remnant of Ner'zhul
		[354388] = 2445, -- Soulrender Dormazain
		[354389] = 2443, -- Painsmith Raznal
		[354390] = 2446, -- Guardian of the First Ones
		[354391] = 2447, -- Fatescribe Roh-Kalo
		[354392] = 2440, -- Kel'Thuzad
		[354393] = 2441, -- Sylvanas Windrunner
	}
```

**Comment by axelkic** [comment]
Also, this is invalid:
```
		-- Sepulcher of the First Ones
		[354390] = 2458, -- Guardian of the First Ones
		[367121] = 2465, -- Skolex, the Insatiable Ravener
		[367124] = 2470, -- Artificer Xy'mox
		[367126] = 2459, -- Dausegne, the Fallen Oracle
		[367128] = 2460, -- Prototype Pantheon
		[367130] = 2461, -- Lihuvim, Principal Architect
		[367132] = 2463, -- Halondrus the Reclaimer
		[367134] = 2469, -- Anduin Wrynn
		[367136] = 2457, -- Lords of Dread
		[367140] = 2467, -- Rygelon
		[367143] = 2464, -- The Jailer, Zovaal
```

There's already an id of `354390` referring to the SoD Guardian of the First Ones, the first Sepulcher can't have the same id. It should start with `367...` like the others

**Comment by fraerie** [comment]
The encounter ID references came from here:
https://wowpedia.fandom.com/wiki/JournalEncounterID

The vantus run IDs came from wowheads DB.

If you have a better reference for Guardian, great, I couldn't find one. I haven't tested Guardian as we didn't need runes for it. The code fragment above has worked for the last four bosses, being the only ones I've tested it for.

**Comment by axelkic** [comment]
I don't have a better reference, I don't know if you even can vantus rune Vigilant Guardian, but we can't have the overlapping id or the code will fail. Going to just remove `Vigilant Guardian` from the list.

**Comment by fraerie** [comment]
The Vigilant Guardian rune wasn't created at the same time as the rest of them

https://www.wowhead.com/spells/uncategorized/name:Vantus+Rune:?filter=21:21;2:5;90105:90205

I recall it taking quite a bit of digging to find it.

**Comment by axelkic** [comment]
I think we want the ones that have the slider: https://www.wowhead.com/spell=359893/vantus-rune-progenitor-defense-system -- but I think Vigilant Guardian === Progenitor

**Comment by fraerie** [comment]
Checking the attributes I concur on 
https://www.wowhead.com/spell=359893/vantus-rune-progenitor-defense-system

But it's the one that isn't aligned with the rest of them for how it's built.

**Comment by axelkic** [comment]
> Checking the attributes I concur on https://www.wowhead.com/spell=359893/vantus-rune-progenitor-defense-system
> 
> But it's the one that isn't aligned with the rest of them for how it's built.
> 
> Ultimately I doubt it's terribly relevant at this point with less than two weeks to the end of the season. People don't really use rune on the first boss of a tier.

True, just for completion sake. Going to look into oils next.

**Comment by TreizeEU** [comment]
We might still use vantus runes in Sepulcher in season 4's fated raid difficulty, so it might not be completely irrelevant quite yet?

And I confirm you can use the vantus rune on first boss by using it on the one attackable npc (Vigilant Custodian?) that's patrolling around, so the number "359893" are correct for Vigilant Guardian. I just ran in with my mage while in a raid group for a world boss, ran past trash and invis'ed. The buff does say "Vantus Rune: Progenitor Defense System".

**Comment by TreizeEU** [comment]
I haven't seen anyone update this addon for a while now, so I'm just going to leave this Consumables.lua file here. Which, I believe, fixes this ticket, as well as ticket 108 for the Eternal Augment Rune buff tracking.

[Consumables.zip](https://github.com/BigWigsMods/oRA3/files/9379283/Consumables.zip)

**Comment by fraerie** [comment]
Note - the updated version of this file has not been included in the 9.2.2 release.

**Comment by funkydude** [comment]
> Note - the updated version of this file has not been included in the 9.2.2 release.

I merged the #112 pull request, thanks!
